src/integration-test/java/com/commercetools/sync/integration/externalsource/products/ProductSyncWithReferencedProductsIT.java

```diff
@@ -0,0 +1,470 @@
+    static void setup() {
+        deleteProductSyncTestData(CTP_TARGET_CLIENT);
+        productType = createProductType(PRODUCT_TYPE_RESOURCE_PATH, CTP_TARGET_CLIENT);
+    }
+
+    /**
+     * Deletes Products from the target CTP project, then it populates target CTP project with product test
+     * data.
+     */
+    @BeforeEach
+    void setupTest() {
+        clearSyncTestCollections();
+        deleteAllProducts(CTP_TARGET_CLIENT);
+        syncOptions = buildSyncOptions();
+
+        final ProductDraft productDraft = ProductDraftBuilder
+            .of(productType, ofEnglish("foo"), ofEnglish("foo-slug"), emptyList())
+            .key("foo")
+            .build();
+
+        final ProductDraft productDraft2 = ProductDraftBuilder
+            .of(productType, ofEnglish("foo2"), ofEnglish("foo-slug-2"), emptyList())
+            .key("foo2")
+            .build();
+
+        product = executeBlocking(CTP_TARGET_CLIENT.execute(ProductCreateCommand.of(productDraft)));
+        product2 = executeBlocking(CTP_TARGET_CLIENT.execute(ProductCreateCommand.of(productDraft2)));
+    }
+
+    private void clearSyncTestCollections() {
+        errorCallBackMessages = new ArrayList<>();
+        errorCallBackExceptions = new ArrayList<>();
+        warningCallBackMessages = new ArrayList<>();
+    }
+
+    private ProductSyncOptions buildSyncOptions() {
+        final Consumer<String> warningCallBack = warningMessage -> warningCallBackMessages.add(warningMessage);
+
+        return ProductSyncOptionsBuilder.of(CTP_TARGET_CLIENT)
+                                        .errorCallback(this::collectErrors)
+                                        .warningCallback(warningCallBack)
+                                        .build();
+    }
+
+    private void collectErrors(final String errorMessage, final Throwable exception) {
+        errorCallBackMessages.add(errorMessage);
+        errorCallBackExceptions.add(exception);
+    }
+
+    @AfterAll
+    static void tearDown() {
+        deleteProductSyncTestData(CTP_TARGET_CLIENT);
+    }
+
+    @Test
+    void sync_withProductReferenceAsAttribute_shouldCreateProductReferencingExistingProduct() {
+        // preparation
+        final AttributeDraft productReferenceAttribute =
+            AttributeDraft.of("product-reference", Reference.of(Product.referenceTypeId(), product.getKey()));
+        final ProductVariantDraft masterVariant = ProductVariantDraftBuilder
+            .of()
+            .sku("sku")
+            .key("new-product-master-variant")
+            .attributes(productReferenceAttribute)
+            .build();
+
+        final ProductDraft productDraftWithProductReference = ProductDraftBuilder
+            .of(productType, ofEnglish("productName"), ofEnglish("productSlug"), masterVariant)
+            .key("new-product")
+            .build();
+
+
+        // test
+        final ProductSync productSync = new ProductSync(syncOptions);
+        final ProductSyncStatistics syncStatistics =
```

Comment:
could you also check the update action in test ?

Reply:
5ad7c95